src/app/auth/auth.component.html

```diff
@@ -1,20 +1,16 @@
-<div *ngIf="!isReady" class="centralize">
-  <mat-spinner [diameter]="47"></mat-spinner>
-</div>
-
-<div *ngIf="isReady && isAppOutdated" class="centralize">
+<div *ngIf="isAppOutdated" class="centralize">
   <span class="error-text"> You are using an outdated version of CATcher. <br/>
     Please download the latest version of CATcher <a [href]="appService.latestReleaseUrl"> here</a>.
   </span>
 </div>
 
-<div *ngIf="isReady && versionCheckingError" class="centralize">
+<div *ngIf="versionCheckingError" class="centralize">
   <span class="error-text"> Unable to verify CATcher's version. <br/>
     Please click <a href="#" (click)="checkAppIsOutdated()">here</a> to try again
   </span>
 </div>
 
-<div *ngIf="isReady && !versionCheckingError && !isAppOutdated">
+<div *ngIf="!versionCheckingError && !isAppOutdated">
```

Comment:
I think this is state representation is iffy. At the initial stage we cannot say that there is "no version checking error" or the "app is not outdated", rather there is "no version checking yet" and "we don't know if app is outdated".
The semantic meaning cannot really be captured by an arbitrary collection of booleans, however, I don't have concrete idea to solve this. Ideally, we want to make use of some kind of Algebraic Data Type to represent the application state but I think Typescript and Angular Templates don't really support this well so I'll just leave this here as food for thought.

Reply:
I agree with you wholeheartedly. 

Maybe we can abstract this logic under a `helper getter`, something to capture the idea of: 
"no errors have been encountered yet on the landing page" , something like 
`hasNoVersionErrors()` or `isLandingPageStable` 

```typescript 
const hasNoVersionErrors(): boolean {
    return !versionCheckingError && !isAppOutdated;
}
```

then we can do something like 
```html 
<div *ngIf="hasNoVersionErrors()">
  ...
</div> 
```

What are your thoughts on this? >< @dingyuchen

Reply:
I was thinking more of something like [this](https://dev.to/mikeskoe/algebraic-data-type-adt-with-typescript-f3p), where we leverage the type system in order to present the state. 

This would let us make the template more declarative, ie from

```
<div *ngIf="hasNoVersionErrors()">
  <div *ngIf="authenticated()">
    ...
  </div>
  <div *ngIf="!authenticating()">
    ...
  </div>
  ...
</div> 
```
to
```
<div *ngIf="stage1">
    ...
</div>
<div *ngIf="stage2">
  ...
</div>
```

And also, we reduce the chances of a regression bug when someone inadvertently produces an invalid state. ie (isSettingUpSession === true and isAppOutdated === true).
Right now this is done programmatically with control flow branches but we can never be sure if some edge case is missed.

With that said I think this is a lot of work for a minor enhancement so we can open a new issue for this.

Reply:
Ah I see, yea I think we adopt this in another PR! 
Sounds pretty good to use sia haha ><

Reply:
Do you mind opening an issue for this after this PR is merged? >< @dingyuchen